src/qt/overviewpage.cpp

```diff
@@ -163,25 +182,25 @@ void OverviewPage::setBalance(const interfaces::WalletBalances& balances)
     m_balances = balances;
     if (walletModel->wallet().isLegacy()) {
         if (walletModel->wallet().privateKeysDisabled()) {
-            ui->labelBalance->setText(BitcoinUnits::formatWithUnit(unit, balances.watch_only_balance, false, BitcoinUnits::separatorAlways));
```

Comment:
Could we simplify this? Maybe define a `class AmountLabel : public QLabel` and here you would just call `setAmount(...)`. And in `setWalletModel` you could `ui->labelBalance->setOptionsModel`. This way `AmountLabel` would call `setText` when privacy or display unit changes (and amount ofc).

Reply:
Removing a tailored widget was [considered](https://github.com/bitcoin/bitcoin/pull/16432#issuecomment-570813848) as simplification already :)
> As @luke-jr rightly [commented](https://github.com/bitcoin/bitcoin/pull/16432#discussion_r362976691):
> > I'm not sure there's a point to a dedicated widget anymore...
>
> So the dedicated `BitcoinAmountLabel` widget has been removed.

Anyway, let's focus this PR on user-faced changes, and leave internal optimization for followup pulls, if you don't mind :)